modules/tinymce/src/core/main/ts/api/EditorUpload.ts

```diff
@@ -173,8 +149,8 @@ const EditorUpload = (editor: Editor): EditorUpload => {
           };
         });
 
-        if (filteredResult.length > 0) {
-          changeHandler.fireIfChanged();
+        if (shouldDispatchChange) {
+          editor.undoManager.dispatchChange();
         }
 
         if (imagesToRemove.length > 0 && !Rtc.isRtc(editor)) {
```

Comment:
This doesn't solve the issue I've repeatedly said here if we're going with the single event logic... this is going to end up with 2 change events when you have both a URL changed and an image that's going to be removed.

Reply:
We also should make sure we have a test whereby 2 images are uploaded and one image succeeds in uploading and the other fails with the `remove` flag set. In that case, given what's been said here then we should only have the single `change` event fire.

Reply:
changed, we can continue this discussion [here](https://github.com/tinymce/tinymce/pull/7840#discussion_r900071060)